src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,123 @@
+        assertThat(Arrays.stream(methods)
+                .map(Method::getReturnType)
+                .map(Class::getName)).containsExactlyInAnyOrder(void.class.getName(), String.class.getName(), String.class.getName());
+    }
+
+    @Test
+    @DisplayName("test로 시작하는 메소드 실행")
+    void testMethodRun() throws Exception {
+        Class<Car> CAR_CLAZZ = Car.class;
+        Car car = CAR_CLAZZ.getConstructor().newInstance();
+        Method[] allMethods = CAR_CLAZZ.getDeclaredMethods();
+        List<Method> testPrefixMethods = Arrays.stream(allMethods)
+                .filter(x -> x.getName().contains("test"))
+                .collect(Collectors.toList());
+        testPrefixMethods
+                .forEach(x -> {
+                    try {
+                        x.invoke(car);
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+    }
+
+    @Test
+    @DisplayName("@PrintView 애노테이션 메소드 실행")
+    void testAnnotationMethodRun() throws Exception {
+        Car car = CAR_CLAZZ.getConstructor(String.class, int.class)
+                .newInstance("name", 1000);
+        Method[] allMethods = CAR_CLAZZ.getDeclaredMethods();
+
+        List<Method> printViewMethods = Arrays.stream(allMethods)
+                .filter(x -> x.isAnnotationPresent(PrintView.class))
+                .collect(Collectors.toList());
+
+        printViewMethods.forEach(x -> {
+            try {
+                x.invoke(car);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
+    @DisplayName("private field에 값 할당")
+    public void privateFieldAccess() throws Exception {
+
+        String name = "name";
+        int price = 1000;
+
+        Car car = CAR_CLAZZ.getConstructor().newInstance();
+        Field nameField = CAR_CLAZZ.getDeclaredField("name");
+        nameField.setAccessible(true);
+        nameField.set(car, name);
+
+        Field priceField = CAR_CLAZZ.getDeclaredField("price");
+        priceField.setAccessible(true);
+        priceField.set(car, price);
+
+        assertThat(getCarFieldValue("name", String.class, car)).isEqualTo(name);
+        assertThat(getCarFieldValue("price", int.class, car)).isEqualTo(price);
+    }
+
+    @Test
+    @DisplayName("인자를 가진 생성자의 인스턴스 생성")
+    void constructorWithArgs() throws Exception {
+        Class<Car> CAR_CLAZZ = Car.class;
+        String name = "name";
+        int price = 1000;
+
+        Constructor<?> constructor = CAR_CLAZZ.getDeclaredConstructor(String.class, int.class);
+
+        Car car = (Car) constructor.newInstance(name, price);
+
+        assertThat(getCarFieldValue("name", String.class, car)).isEqualTo(name);
+        assertThat(getCarFieldValue("price", int.class, car)).isEqualTo(price);
+    }
+
+    private <T> T getCarFieldValue(String fieldName, Class<T> fieldType, Car carInstance) throws Exception {
```

Comment:
```suggestion
    private <T> T getCarFieldValue(String fieldName, Car carInstance) throws Exception {
```

필드 타입은 사용하고 있지 않는 것 같아요 :)